keytree/src/lib.rs

```diff
@@ -48,6 +52,35 @@ impl Xprv {
     }
 }
 
+impl Xpub {
+    /// Returns a intermediate child pubkey.
+    pub fn to_intermediate_key(&self, customize: impl FnOnce(&mut Transcript)) -> Xpub {
```

Comment:
`to_***` is used for converting type A to type B. Here we do not convert, we derive child keys, leaving the parent happily alone.

Suggestion:
```suggestion
    pub fn derive_intermediate_key(&self, customize: impl FnOnce(&mut Transcript)) -> Xpub {
```